backend/src/composite/mod.rs

```diff
@@ -92,6 +140,7 @@ struct MachineData<'a, F> {
 
 pub(crate) struct CompositeBackend<'a, F> {
     machine_data: BTreeMap<String, MachineData<'a, F>>,
+    machine_names: Vec<String>,
```

Comment:
Yes, because `self.machine_data` is a `BTreeMap`, i.e., keys are ordered. The reason I introduced it was because the ordering matters, but I just removed it again! :)